backend/src/main/scala/com/softwaremill/bootzooka/util/SecureRandomId.scala

```diff
@@ -0,0 +1,48 @@
+package com.softwaremill.bootzooka.util
+
+import cats.effect.Sync
+import java.security.SecureRandom
+import org.apache.commons.codec.binary.Hex
+
+/** copy-pasted from TSec https://github.com/jmcardon/tsec */
+
+/** A trait that manages a secureRandom instance.
+  */
+trait ManagedRandom {
+
+  /** Cache our random, and seed it properly as per
+    * [[https://tersesystems.com/2015/12/17/the-right-way-to-use-securerandom/]]
+    */
+  private val cachedRand: SecureRandom = {
+    val r =
+      SecureRandom.getInstance(if (scala.util.Properties.isWin) ManagedRandom.WinRandom else ManagedRandom.UnixURandom)
+    r.nextBytes(new Array[Byte](20)) //Force reseed
+    r
+  }
+
+  def nextBytes(bytes: Array[Byte]): Unit =
+    cachedRand.nextBytes(bytes)
+}
+
+object ManagedRandom {
+  private[ManagedRandom] val WinRandom   = "Windows-PRNG"
+  private[ManagedRandom] val UnixURandom = "NativePRNGNonBlocking"
+}
+
+object SecureRandomId {
+  lazy val Strong: SecureRandomIdGenerator      = SecureRandomIdGenerator(32)
+  lazy val Interactive: SecureRandomIdGenerator = SecureRandomIdGenerator(16)
+
+  def apply(s: String): SecureRandomId  = s.asInstanceOf[SecureRandomId]
+  def coerce(s: String): SecureRandomId = s.asInstanceOf[SecureRandomId]
+}
+
+case class SecureRandomIdGenerator(sizeInBytes: Int = 32) extends ManagedRandom {
+  def generate: SecureRandomId = {
+    val byteArray = new Array[Byte](sizeInBytes)
+    nextBytes(byteArray)
+    new String(Hex.encodeHex(byteArray)).asInstanceOf[SecureRandomId]
```

Comment:
I think that java 11+ (which we are using) has a hex-encoding method somewhere? Maybe we can then get rid of one dependency

Reply:
Hmm... Java 17 introduces ([HexFormat](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/HexFormat.html)), still - it is possible to do this conversion using `String.format()` like this:

```
byteArray.map(b => String.format("%02x", b)).mkString
```

I would change it and remove dependecy.